flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorUtils.java

```diff
@@ -64,16 +64,18 @@
 	private static final Logger LOG = LoggerFactory.getLogger(WebMonitorUtils.class);
 
 	/**
-	 * Singleton to hold the log and stdout file.
+	 * Singleton to hold the log file, the stdout file, the log directory.
 	 */
 	public static class LogFileLocation {
 
 		public final File logFile;
 		public final File stdOutFile;
+		public final String logDir;
```

Comment:
Why did you chose `String` to represent the `logDir`? `java.io.File` can also represent directories.

Reply:
That sounds good. As I before used filePath, so I defined it a string.